packages/batch-execute/src/mergeRequests.ts

```diff
@@ -117,12 +117,18 @@ function prefixRequest(prefix: string, request: ExecutionRequest): ExecutionRequ
   let prefixedDocument = aliasTopLevelFields(prefix, request.document);
 
   const executionVariableNames = Object.keys(executionVariables);
+  const hasFragmentDefinitions = request.document.definitions.some(def => isFragmentDefinition(def));
+  const fragmentSpreadImpl: Record<string, boolean> = {};
 
-  if (executionVariableNames.length > 0) {
+  if (executionVariableNames.length > 0 || hasFragmentDefinitions) {
```

Comment:
@yaacovCR – have a look at 44ba4cb8d6565d2729da3145b98a097f290b2af1. I've fixed to bugs related to fragments here:

1. Fragments weren't getting prefixed unless there were also variables.
2. Inlined root fragments were possibly leaving an invalid orphaned fragment definition. This now filters fragment definitions out unless the have a surviving spread implementation.